src/main/java/core/basesyntax/strategy/DataConverterImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.FruitTransaction.Action;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    public List<FruitTransaction> convertToTransaction(List<String> dateList) {
```

Comment:
The method `convertToTransaction` should validate the input data before processing it. Ensure that the `dateList` is not null or empty, and that each string in the list has the correct format and contains the expected number of elements before splitting and parsing it.